runner/src/main.rs

```diff
@@ -659,10 +659,10 @@ fn run_example(opt: &RunExamples, example: &Example) -> Step {
     let run_backend_server_clients: Step = if opt.run_server.unwrap_or(true) {
         let run_server_clients = if example.applications.is_empty() {
             if opt.build_client.client_variant == NO_CLIENTS {
-                panic!(
-                    "`{}` client variant is not supported when no applications are provided",
-                    NO_CLIENTS
-                );
+                Step::Multiple {
+                    name: "run clients (empty)".to_string(),
+                    steps: vec![],
+                }
```

Comment:
Why is this special case necessary? Wouldn't `run_clients` be naturally empty already?

Reply:
Done